sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -0,0 +1,243 @@
+        item in `ax`. Only plots with a two feature will be populated in
+        `contours_`. Line plots are stored in `lines_`.
+
+    figure_ : matplotlib Figure
+        Figure containing partial dependence plots.
+    """
+    def __init__(self, pd_results, features, feature_names, target_idx,
+                 pdp_lim, deciles):
+        self.pd_results = pd_results
+        self.features = features
+        self.feature_names = feature_names
+        self.target_idx = target_idx
+        self.pdp_lim = pdp_lim
+        self.deciles = deciles
+
+    def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None, fig=None):
+        """Plot partial dependence plots.
+
+        Parameters
+        ----------
+        ax : Matplotlib axes, list of Matplotlib axes or None, default=None
+            Axes to plot the partial dependence curves. If a single axes is
+            given, it is treated as a bounding axes and a grid of partial
+            depdendence plots will be drawn on that top of it. If a list of
+            axes are passed, the partial dependence plots will be drawn on
+            those axes. By default, a single bounding axes is created and
+            treated as the single axes case.
+
+        n_cols : int, default=3
+            The maximum number of columns in the grid plot.
+
+        line_kw : dict or None, default=None
+            Dict with keywords passed to the ``matplotlib.pyplot.plot`` call.
+            For one-way partial dependence plots.
+
+        contour_kw : dict or None, default=None
+            Dict with keywords passed to the ``matplotlib.pyplot.contourf``
+            call for two-way partial dependence plots.
+
+        fig : Matplotlib figure object or None, default=None
+            A figure object onto which the plots will be drawn, after the
+            figure has been cleared. By default, a new one is created.
+
+            .. deprecated:: 0.22
+
+        Returns
+        -------
+        display: :class:`~sklearn.inspection.PartialDependenceDisplay`
+        """
+
+        check_matplotlib_support("plot_partial_dependence")
+        import matplotlib.pyplot as plt  # noqa
+        from matplotlib import transforms  # noqa
+        from matplotlib.ticker import MaxNLocator  # noqa
+        from matplotlib.ticker import ScalarFormatter  # noqa
+        from matplotlib.gridspec import GridSpecFromSubplotSpec  # noqa
+
+        if line_kw is None:
+            line_kw = {}
+        if contour_kw is None:
+            contour_kw = {}
+
+        contour_kw = {**{"alpha": 0.75}, **contour_kw}
+
+        if fig is not None:
+            warnings.warn("The fig parameter is deprecated in version "
+                          "0.22 and will be removed in version 0.24",
+                          DeprecationWarning)
+            fig.clear()
+
+        if ax is None:
+            if fig is None:
+                _, ax = plt.subplots()
+            else:
+                ax = fig.add_subplot(111)
```

Comment:
I hate the matlab syntax. I would use either ``add_subplot(1, 1, 1)`` or ``gca``?